src/main/java/io/nats/streaming/StreamingConnectionImpl.java

```diff
@@ -552,14 +552,25 @@ protected void processAck(Message msg) {
     TimerTask createAckTimerTask(String guid) {
         TimerTask task = new java.util.TimerTask() {
             public void run() {
-                processAckTimeout(guid);
+                try {
+                    processAckTimeout(guid);
+                } catch (Exception e) {
+                    // catch exception to prevent the timer to be closed
+                    logger.error("stan: error encountered during processAckTimeout, will cancel this timer task", e);
+                    // cancel this task
+                    cancel();
+                }
             }
         };
         return task;
     }
 
     protected void processAckTimeout(String guid) {
         AckClosure ackClosure = removeAck(guid);
+        if (ackClosure == null) {
```

Comment:
I observed while debugging the `ITConnectionTest::testRaceAckOnClose`, that `ackClosure` returned from `removeAck` if waiting long enough in the breakpoint was `null`, and `NPE` was thrown by the line below.